modules/GKB/CommonUtils.pm

```diff
@@ -321,22 +319,26 @@ sub XOR {
     return ($expression1 || $expression2) && (!($expression1 && $expression2));
 }
 
-sub get_components {
-    my $composite_entity = shift;
-    
-    my @components;
-    foreach my $component (@{$composite_entity->hasComponent},
-                           @{$composite_entity->hasMember},
-                           @{$composite_entity->hasCandidate},
-                           @{$composite_entity->repeatedUnit}) {
-        push @components, $component;
-        my @sub_components = grep {defined} get_components($component);
-        if (@sub_components) {
-            push @components, @sub_components;
-        }
-    }
-    
-    return @components;
+sub get_components
+{
+	my $composite_entity = shift;
+
+	my @components;
+	if ($composite_entity)
+	{
+		foreach my $component (@{$composite_entity->hasComponent}, @{$composite_entity->hasMember},
+								@{$composite_entity->hasCandidate}, @{$composite_entity->repeatedUnit})
+		{
+			push @components, $component;
+			my @sub_components = grep {defined} get_components($component);
+			if (@sub_components)
+			{
```

Comment:
This might need to be evaluated as 'if $sub_components[0]'

Reply:
Maybe? I just changed the formatting here. I'm not sure if `if ($sub_components[0]) {...` is better than `if (@sub_components) {` 
Is it possible there are cases where one would evaluate to TRUE and the other to FALSE?